examples/vqe_toric.py

```diff
@@ -0,0 +1,609 @@
+            edge_idx = i * Ly + (j - 1)
+            connected_edges.append(edge_idx)
+
+        # Right horizontal edge
+        if j < Ly:
+            edge_idx = i * Ly + j
+            connected_edges.append(edge_idx)
+
+        if connected_edges:
+            pauli_str = ["I"] * num_qubits
+            for idx in connected_edges:
+                pauli_str[idx] = "X"
+            pauli_str = "".join(pauli_str)
+
+            # Adjust weight for boundary vertices
+            weight = 1.0
+            if (i == 0 or i == Lx) and (j == 0 or j == Ly):
+                weight = 0.5  # Corner vertices
+            elif i == 0 or i == Lx or j == 0 or j == Ly:
+                weight = 0.75  # Edge vertices
+
+            H[pauli_str] = -weight * (1 - h)
+
+    # Plaquette operators B_p (Z products on edges around each plaquette)
+    for i, j in product(range(Lx), range(Ly)):
+        top = i * Ly + j
+        bottom = (i + 1) * Ly + j
+        left = num_horizontal + i * (Ly + 1) + j
+        right = left + 1
+
+        pauli_str = ["I"] * num_qubits
+        for idx in [top, bottom, left, right]:
+            pauli_str[idx] = "Z"
+        pauli_str = "".join(pauli_str)
+
+        H[pauli_str] = -(1 - h)
+
+    # External field terms
+    for q in range(num_qubits):
+        if hx != 0:
+            x_str = ["I"] * num_qubits
+            x_str[q] = "X"
+            H["".join(x_str)] = -hx * h
+
+        if hz != 0:
+            z_str = ["I"] * num_qubits
+            z_str[q] = "Z"
+            H["".join(z_str)] = -hz * h
+
+    print(f"Hamiltonian built: {len(H)} terms")
+    return H
+
+
+def get_plaquette_qubits(i: int, j: int, Lx: int, Ly: int) -> Dict[str, int]:
+    """
+    Get qubit indices for the four edges of plaquette (i,j).
+
+    Args:
+        i, j: Plaquette coordinates
+        Lx, Ly: Lattice dimensions
+
+    Returns:
+        Dictionary with keys 'top', 'bottom', 'left', 'right' mapping to qubit indices
+    """
+    num_horizontal = (Lx + 1) * Ly
+
+    return {
+        "top": int(i * Ly + j),
+        "bottom": int((i + 1) * Ly + j),
+        "left": int(num_horizontal + i * (Ly + 1) + j),
+        "right": int(num_horizontal + i * (Ly + 1) + j + 1),
+    }
+
+
+def building_block(circuit: tc.Circuit, params: tf.Variable, q1: int, q2: int) -> None:
+    """
+    Apply a parameterized two-qubit gate sequence.
+
+    Gate sequence: RZ-RY-RZ-RZ-RXX-RYY-RZZ-RZ-RY-RZ-RZ
+
+    Args:
+        circuit: Quantum circuit
+        params: 11 parameters for the gate sequence
+        q1, q2: Qubit indices
+    """
+    circuit.RZ(q1, theta=params[0])
+    circuit.RY(q2, theta=params[1])
+    circuit.RZ(q1, theta=params[2])
+    circuit.RZ(q2, theta=params[3])
+    circuit.RXX(q1, q2, theta=params[4])
+    circuit.RYY(q1, q2, theta=params[5])
+    circuit.RZZ(q1, q2, theta=params[6])
+    circuit.RZ(q1, theta=params[7])
+    circuit.RY(q2, theta=params[8])
+    circuit.RZ(q1, theta=params[9])
+    circuit.RZ(q2, theta=params[10])
```

Comment:
how this is su4 gate as su4 gate has 15 parameters? though it is ok for demonstration here, the latest version of tc has native c.su4(i, i+1, theta=vec_15).